pkg/services/pluginsintegration/plugininstaller/service_test.go

```diff
@@ -232,8 +248,12 @@ func TestService_Run(t *testing.T) {
 						expectedInstalled++
 					}
 				}
-				require.Equal(t, expectedInstalled, installed)
-				require.Equal(t, expectedInstalledFromURL, installedFromURL)
+				require.Eventually(t, func() bool {
+					return installed == expectedInstalled
+				}, time.Second, time.Millisecond*10)
+				require.Eventually(t, func() bool {
+					return installedFromURL == expectedInstalledFromURL
+				}, time.Second, time.Millisecond*10)
```

Comment:
Are you worried about flakiness here? I wonder should we add a channel to the `running` func that tests can use to determine when to safely verify install state. Not ideal just for the sake of a test perhaps but would simplify the tests 🤷